packages/quantic/force-app/main/default/lwc/quanticDateFacet/quanticDateFacet.js

```diff
@@ -53,7 +53,10 @@ export default class QuanticDateFacet extends LightningElement {
   }
 
   get values() {
-    return this.state.values || [];
+    return this.state.values.map(v => ({
+      ...v,
+      checked: v.state === 'selected'
```

Comment:
I'd use the term `selected` instead of `checked` since it's the term used in headless for that concept.

Reply:
checked refers to the checkbox state in my point of view, not the facet value being selected. But it seems like tomatoes, tomAtoes to me. I don't really mind either way.

Reply:
following that logic, if the checkbox is checked but the value is selected, then given this refers to the value maybe selected is indeed the better term?

Reply:
Well it's really `get contentToDisplay()` which returns an array of content to display, for instance it may contain `begin/end` etc. and other data, so it is an object really and not just a facet value if we want to use that definition semantically.

As I said, I can change it, but to me it's way clearer that this value means it will be used as a checked attribute of a checkbox rather than that the facet value (a kinda abstract concept) is selected.

Selected doesn't **say** what it does, while checked IMO is clearer as to what the effect is.

Reply:
> ...but to me it's way clearer that this value means it will be used as a checked attribute of a checkbox rather than that the facet value...

`checked` refers to the actual implementation of the facet, and that is the problem. The facet implementation could change to not use checkboxes. It this happens, then having attributes like `is-checked` would not make sense anymore, but changing it would then be breaking.

So I think `selected` is a better choice as it is more abstract and give us more room to change the implementation in the future.

Does it make sense to you?

Reply:
That's the whole point for me, we ARE building the actual implementation of the facet as components.

If we were to change the facet value component in the future to not include a checkbox, as a user of that component I would be very upset and I would've expected there to be a different component instead of changing the actual one that displays a checkbox and a label.

But I will change it if that's what's blocking the PR from being approved ;)

Reply:
I get your point, especially about custom styling. I still think `checked` refers to something internal to the component that shouldn't be exposed through its API (state, attributes, etc).

We should take some time to discuss offline how we want the quantic components to evolve. I understand why you suggest to create a new component, but as soon as you have a component that is more than 1-level deep, you'll have to duplicate the whole component tree to maintain the compatibility. Typically this causes lots of duplication and also becomes confusing to users as they don't know which component to use.

Reply:
Will be fixed by the redesign of the facets in a later Jira.